### PR TITLE
[MM-28150] e2e: add MM-T924, MM-T928, MM-T929, MM-T930

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Adds automated test cases for the first half of the ticket: 

- [MM-T924](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T924)
- [MM-T928](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T928)
- [MM-T929](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T929)
- [MM-T930](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T930)

PS: It got quite big already, I thought it would be better to split into two PRs.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-28150

### CODE REVIEW

File: e2e/cypress/integration/system_console/user_management_spec.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+            cy.get('#searchableUserListNextBtn').should('be.visible').click();
+
+            // * Count at top changes appropriately.
+            cy.get('#searchableUserListTotal').then((el2) => {
+                const count2 = el2[0].innerText.replace(/\n/g, '').replace(/\s/g, ' ');
+                expect(count1).not.equal(count2);
+            });
+
+            // * Can page backward as well.
+            cy.get('button[id="searchableUserListPrevBtn"]').should('be.visible').click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/user_management_spec.js:43]: nit
```suggestion
            cy.get('#searchableUserListPrevBtn').should('be.visible').click();
```

File: e2e/cypress/integration/system_console/user_management_spec.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        // # Search for the user.
+        cy.get('#searchUsers').clear().type(oldEmail).wait(TIMEOUTS.HALF_SEC);
+
+        // # Open the actions menu.
+        cy.get('[data-testid="userListRow"] .more-modal__right .more-modal__actions .MenuWrapper .text-right a').
+            click().wait(TIMEOUTS.HALF_SEC);
+
+        // # Click the Update email menu option.
+        cy.get('[data-testid="userListRow"] .more-modal__right .more-modal__actions .MenuWrapper .MenuWrapperAnimation-enter-done').
+            find('li').eq(4).click().wait(TIMEOUTS.HALF_SEC);
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/user_management_spec.js:160]: I'm good as is but could be improved to make less susceptible to CSS/DOM structure change in the future.
```javascript
        cy.findByTestId('userListRow').within(() => {
            // # Open the actions menu.
            cy.findByText('Member').click().wait(TIMEOUTS.HALF_SEC);

            // # Click the Update email menu option.
            cy.findByLabelText('User Actions Menu').findByText('Update Email').click().wait(TIMEOUTS.HALF_SEC);
        });
```

File: e2e/cypress/integration/system_console/user_management_spec.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+        // # Click the "Reset" button.
+        cy.findByTestId('resetEmailButton').click();
+
+        // * Check for the error messages, if any.
+        if (errorMsg.length > 0) {
+            cy.get('form.form-horizontal').find('.has-error p.error').should('be.visible').and('contain', errorMsg);
+
+            // # Close the modal.
+            cy.get('button[aria-label="Close"]').click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/user_management_spec.js:178]: nit
```suggestion
            cy.findByLabelText('Close').click();
```

**Review by saturninoabril — APPROVED** [review]
Thanks @isacikgoz! Overall looks good, tested and passed. Left non-blocking comments.

**Review by streamer45 — APPROVED** [review]
Nicely done :+1:

**Review by lindy65 — APPROVED** [review]
Thanks @isacikgoz - looks good to me 👍

**Pull request merged**